docs/scripts/readme_plot.py

```diff
@@ -67,45 +79,48 @@ class MoneyModelDF(ModelDF):
     def __init__(self, N):
         super().__init__()
         self.num_agents = N
-        self.create_agents(N, {MoneyAgentDF: 1})
+        self.agents = self.agents.add(MoneyAgentsDF(N, model=self))
+
+    def step(self):
+        self.agents = self.agents.do("step")
 
-    def step(self, merged_mro=True):
-        self.agents = self.agents.sample(frac=1)
-        self.update_agents_masks()
-        super().step(merged_mro)
+    def run_model(self, n):
+        for _ in range(n):
+            self.step()
 
 
-class MoneyAgentDF(AgentDF):
-    dtypes: dict[str, str] = {"wealth": "int64"}
+class MoneyAgentsDF(AgentSetDF):
+    def __init__(self, n: int, model: MoneyModelDF):
+        super().__init__(model=model)
+        self.add(n, data={"wealth": np.ones(n)})
 
-    @classmethod
-    def __init__(cls):
-        super().__init__()
-        cls.model.agents.loc[cls.mask, "wealth"] = 1
+    def step(self):
+        wealthy_agents = self.agents["wealth"] > 0
+        self.select(wealthy_agents).do("give_money")
 
-    @classmethod
-    def step(cls):
-        wealthy_agents = cls.model.agents.loc[cls.mask, "wealth"] > 0
-        if wealthy_agents.any():
-            other_agents = cls.model.agents.index.isin(
-                cls.model.agents.sample(n=wealthy_agents.sum()).index
-            )
-            cls.model.agents.loc[wealthy_agents, "wealth"] -= 1
-            cls.model.agents.loc[other_agents, "wealth"] += 1
+    def give_money(self):
+        other_agents = self.agents.sample(len(self.active_agents), replace=True)
```

Comment:
Isn't the term "active agent" an internal one, given that it is the ones selected by `select`? If so, it'd be confusing to use it as a public API.

Reply:
OK, I think `selected_agents` is clearer, but we can do it in another PR. I think the user is more used to how it is done in pandas/Polars, in that the selected agents are stored as a variable *defined* by the user themselves.

Reply:
We can change the term `active_agents` to `selected_agents` (and also the corresponding string mask to "selected"). The `active_agents` property has a setter so that user can define it directly with a mask, instead of using the `select `method.